axelor-account/src/main/java/com/axelor/apps/account/web/ClosureAssistantLineController.java

```diff
@@ -0,0 +1,142 @@
+import com.axelor.exception.service.TraceBackService;
+import com.axelor.i18n.I18n;
+import com.axelor.inject.Beans;
+import com.axelor.rpc.ActionRequest;
+import com.axelor.rpc.ActionResponse;
+import java.math.BigDecimal;
+import java.util.Map;
+
+public class ClosureAssistantLineController {
+
+  public void validateClosureAssistantLine(ActionRequest request, ActionResponse response) {
+
+    try {
+      ClosureAssistantLine closureAssistantLine =
+          request.getContext().asType(ClosureAssistantLine.class);
+      closureAssistantLine =
+          Beans.get(ClosureAssistantLineRepository.class).find(closureAssistantLine.getId());
+      Beans.get(ClosureAssistantLineService.class)
+          .validateClosureAssistantLine(closureAssistantLine);
+      response.setReload(true);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void cancelClosureAssistantLine(ActionRequest request, ActionResponse response) {
+
+    try {
+      ClosureAssistantLine closureAssistantLine =
+          request.getContext().asType(ClosureAssistantLine.class);
+      closureAssistantLine =
+          Beans.get(ClosureAssistantLineRepository.class).find(closureAssistantLine.getId());
+      Beans.get(ClosureAssistantLineService.class).cancelClosureAssistantLine(closureAssistantLine);
+      response.setReload(true);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void openViewLinkToAction(ActionRequest request, ActionResponse response) {
+
+    try {
+      ClosureAssistantLine closureAssistantLine =
+          request.getContext().asType(ClosureAssistantLine.class);
+
+      Map<String, Object> view =
+          Beans.get(ClosureAssistantLineService.class).getViewToOpen(closureAssistantLine);
+      if (view != null) {
+        response.setView(view);
+      }
+    } catch (Exception e) {
+      TraceBackService.trace(response, e, ResponseMessageType.ERROR);
+    }
+  }
+
+  public void checkNoExistingClosureAssistantForSameYear(
+      ActionRequest request, ActionResponse response) {
+
+    try {
+
+      ClosureAssistantLine closureAssistantLine =
+          request.getContext().asType(ClosureAssistantLine.class);
+      ClosureAssistant closureAssistant = closureAssistantLine.getClosureAssistant();
+      if (closureAssistant.getId() != null) {
+        closureAssistant =
+            Beans.get(ClosureAssistantRepository.class).find(closureAssistant.getId());
+      }
+
+      if (Beans.get(ClosureAssistantService.class)
+          .checkNoExistingClosureAssistantForSameYear(closureAssistant)) {
+        response.setException(
+            new AxelorException(
+                TraceBackRepository.CATEGORY_INCONSISTENCY,
+                I18n.get(IExceptionMessage.ACCOUNT_CLOSURE_ASSISTANT_ALREADY_EXISTS_FOR_SAME_YEAR),
+                closureAssistant.getFiscalYear().getCode(),
+                closureAssistant.getCompany().getCode()));
```

Comment:
I think it will be more standard to use setError and use String.format to format the message, instead of creating an exception.